app/celery/tasks.py

```diff
@@ -323,8 +323,9 @@ def build_dvla_file(self, job_id):
             )
             dao_update_job_status(job_id, JOB_STATUS_READY_TO_SEND)
         else:
-            current_app.logger.info("All notifications for job {} are not persisted".format(job_id))
-            self.retry(queue=QueueNames.RETRY, exc="All notifications for job {} are not persisted".format(job_id))
+            msg = "All notifications for job {} are not persisted".format(job_id)
+            current_app.logger.info(msg)
+            self.retry(queue=QueueNames.RETRY, exc=Exception(msg))
     except Exception as e:
         # ? should this retry?
```

Comment:
We should implement some retry code here. When would it throw an exception? Database throws exception, s3upload throws an exception. Both seem like reasons to retry.

Reply:
self.retry throws the `exc` parameter. 🤔

Reply:
yeah so the exception block should retry - might refactor out the retry on line 328 then.